uspace/drv/bus/usb/ehci/hw_struct/transfer_descriptor.c

```diff
@@ -69,19 +68,20 @@ static const uint32_t dir[] = {
 	[USB_DIRECTION_BOTH] = TD_STATUS_PID_SETUP,
 };
 
+#include <usb/debug.h>
+
 /**
  * Initialize EHCI TD.
  * @param instance TD structure to initialize.
- * @param next Next TD in ED list.
+ * @param next_phys Next TD in ED list.
  * @param direction Used to determine PID, BOTH means setup PID.
  * @param buffer Pointer to the first byte of transferred data.
  * @param size Size of the buffer.
  * @param toggle Toggle bit value, use 0/1 to set explicitly,
  *        any other value means that ED toggle will be used.
  */
-void td_init(td_t *instance, const td_t *next,
-    usb_direction_t direction, const void *buffer, size_t size, int toggle,
-    bool ioc)
+void td_init(td_t *instance, uintptr_t next_phys, uintptr_t buffer,
```

Comment:
Why has the type of `buffer` changed?
If it's a physical address (or other kind of different address space), it would be more readable if this used a typedef describing what kind of pointer it is.

Reply:
Previously, it was a virtual pointer to a generic data, now it's a physical one. uintptr_t is used as physical pointer all over the project...

Reply:
In the future, physical addresses will be able to have different width than virtual addresses (for example when we support PAE on 32-bit systems), but IMO it's out of scope for this project and will need a global change later.

Reply:
I see. Yes, if you use physical pointers a lot, it wouldn't make sense to change just this instance. And it would also need to be consistent with the rest of the system (a special type in `/abi`?). So yeah, out of scope.